components/butterfly/src/rumor/dat_file.rs

```diff
@@ -269,6 +274,9 @@ impl DatFile {
             file.sync_all()
                 .map_err(|err| Error::DatFileIO(self.path.clone(), err))?;
         }
+
+        perm::set_permissions(&tmp_path, DAT_FILE_PERMISSIONS)?;
```

Comment:
This [is no-op on Windows](https://github.com/habitat-sh/core/blob/master/components/core/src/os/filesystem/windows.rs#L32-L34), which is pretty dangerously misleading.

I'd need to do a bit of research or ask @mwrock, but I think we need to fix the `core` crate.

Reply:
On Windows this is essentially a no op. I think its fine to move forward with this PR as is but to apply these restrictions on Windows, you are correct and we would have to change core. The tricky thing with that is that the api of `chmod` will look fundamentally different on windows since its ACL architecture is more complex. However I think we should deal with that separately.

Reply:
I definitely agree that it should be a separate change. I mean, it kinda has to be because it's in `core`, but I'd at least like to see the `set_permissions` (and any other calls that are no-ops) calls inside a `if cfg!(linux)` to make it clear that this doesn't apply to windows. Right now the code is very misleading.